### PR TITLE
Update section on authorization code binding

### PR DESCRIPTION
**Pull request by PieterKas**
Proposed update based on feedback in issue #47 

Open question for reviewers (and @bc_pi) whether we should allow both or only a single method, and if a single methods should we opt for the least common denominator  (dpop_jkt) or an authorization challenge endpoint specific one?

### CODE REVIEW

**Comment by bc-pi** [comment]
Commented over in https://github.com/aaronpk/oauth-first-party-apps/issues/47 that I think the single method of the DPoP proof header is sufficient and preferred.

**Comment by aaronpk** [comment]
I would tend to agree that we should stick with _only_ the header option for DPoP.

File: draft-parecki-oauth-first-party-apps.md [hunk 1]
```diff
@@ -867,6 +869,10 @@ These design decisions should enable authorization server implementations to iso
 
 # Document History
 
+-latest
```
Comment (bc-pi) [inline draft-parecki-oauth-first-party-apps.md:872]: do you want to use `-latest` here? Typically I've seen the coming draft number used, like `-02`, but maybe I don't know about some convention or something...

**Review by bc-pi — APPROVED** [review]
looks okay to me (pending conflicts that need to be resolved)

File: draft-parecki-oauth-first-party-apps.md (hunk 1 above)
Comment (aaronpk) [reply draft-parecki-oauth-first-party-apps.md:872]: I'll update that before I publish

File: draft-parecki-oauth-first-party-apps.md [hunk 2]
```diff
@@ -552,7 +552,9 @@ DPoP ({{RFC9449}}) is an application-level mechanism for sender-constraining OAu
...
+The mechanism for Authorization Code binding with DPoP is similar as that defined for Pushed Authorization Requests (PARs) in Section 10.1 of {{RFC9449}}. In order to bind the Authorization Code with DPoP, the client MUST add the DPoP header to the Authorization Challenge Request. The authorization server MUST check the DPoP proof JWT that was included in the DPoP header as defined in Section 4.3 of {{RFC9449}}. The authorization server must ensure that the same key is used in all subsequent Authorization Challenge Requests, or in the eventual token request. The authorization server MUST reject subsequent Authorization Challenge Requests, or the eventual token request, unless a DPoP proof for the same key presented in the original Authorization Challenge Request is provided.
```
Comment (aaronpk) [inline draft-parecki-oauth-first-party-apps.md:555]:
```suggestion
The mechanism for Authorization Code binding with DPoP is similar as that defined for Pushed Authorization Requests (PARs) in Section 10.1 of {{RFC9449}}. In order to bind the Authorization Code with DPoP, the client MUST add the DPoP header to the Authorization Challenge Request. The authorization server MUST check the DPoP proof JWT that was included in the DPoP header as defined in Section 4.3 of {{RFC9449}}. The authorization server MUST ensure that the same key is used in all subsequent Authorization Challenge Requests, or in the eventual token request. The authorization server MUST reject subsequent Authorization Challenge Requests, or the eventual token request, unless a DPoP proof for the same key presented in the original Authorization Challenge Request is provided.
```

**Review by aaronpk — APPROVED** [review]

**Pull request merged**